prometheus-metrics-ghc/src/Prometheus/Metric/GHC.hs

```diff
@@ -125,13 +134,193 @@ ghcCollectors = [
+            GaugeType
+            max_live_bytes
+    , statsCollector
+            "ghc_max_large_objects_bytes"
+            "Maximum live data in large objects"
+            GaugeType
+            max_large_objects_bytes
+    , statsCollector
+            "ghc_max_compact_bytes"
+            "Maximum live data in compact regions"
+            GaugeType
+            max_compact_bytes
+    , statsCollector
+            "ghc_max_slop_bytes"
+            "Maximum slop"
+            GaugeType
+            max_slop_bytes
+    , statsCollector
+            "ghc_max_mem_in_use_bytes"
+            "Maximum memory in use by the RTS"
+            GaugeType
+            max_mem_in_use_bytes
+    , statsCollector
+            "ghc_cumulative_live_bytes_total"
+            "Sum of live bytes across all major GCs. Divided by major_gcs gives the average live data over the lifetime of the program."
+            CounterType
+            cumulative_live_bytes
+    , statsCollector
+            "ghc_copied_bytes_total"
+            "Sum of copied_bytes across all GCs"
+            CounterType
+            copied_bytes
+    , statsCollector
+            "ghc_par_copied_bytes_total"
+            "Sum of copied_bytes across all parallel GCs"
+            CounterType
+            par_copied_bytes
+    , statsCollector
+            "ghc_cumulative_par_max_copied_bytes_total"
+            "Sum of par_max_copied_bytes across all parallel GCs"
+            CounterType
+            cumulative_par_max_copied_bytes
+    , statsCollector
+            "ghc_mutator_cpu_seconds_total"
+            "Total CPU time used by the mutator"
+            CounterType
+            (rtsTimeToSeconds . mutator_cpu_ns)
+    , statsCollector
+            "ghc_mutator_elapsed_seconds_total"
+            "Total elapsed time used by the mutator"
+            CounterType
+            (rtsTimeToSeconds . mutator_elapsed_ns)
+    , statsCollector
+            "ghc_gc_cpu_seconds_total"
+            "Total CPU time used by the GC"
+            CounterType
+            (rtsTimeToSeconds . gc_cpu_ns)
+    , statsCollector
+            "ghc_gc_elapsed_seconds_total"
+            "Total elapsed time used by the GC"
+            CounterType
+            (rtsTimeToSeconds . gc_elapsed_ns)
+    , statsCollector
+            "ghc_cpu_seconds_total"
+            "Total CPU time (at the previous GC)"
+            CounterType
+            (rtsTimeToSeconds . cpu_ns)
+    , statsCollector
+            "ghc_elapsed_seconds_total"
+            "Total elapsed time (at the previous GC)"
+            CounterType
+            (rtsTimeToSeconds . elapsed_ns)
+
+    , statsCollector
+            "ghc_gcdetails_gen"
+            "The generation number of this GC"
+            HistogramType -- TODO: is this correct?
+                          -- Gauge makes little sense here.
+                          -- With Histogram we'll be able to see which
+                          -- generations are collected most often.
```

Comment:
I don't want to create stop energy, so feel free to disregard this and proceed with merging.

On reflection, we should probably not be exporting this at all, at least, not as it is.

If I understand correctly (a big "if"), there should be a labelled Counter, where the label is the generation number, and the value is the number of GCs for that generation.

Further, the current generation should be applied as a label to things like `ghc_gcdetails_copied_bytes`, which would make the metrics more useful.

Realistically, I think it's fine to release these with the metrics clearly marked as alpha & subject to change without prior deprecation warnings in future releases. Nothing will beat actually using these.

Reply:
Yep, agree with this - fine to push this out and then change it later IMO.